pkg/aws_config_client/completer.go

```diff
@@ -113,14 +113,99 @@ func (c *completer) SurveyProfile() (*AWSNamedProfile, error) {
 		Name:       profileName,
 		AWSProfile: profile,
 	}
+	consumedProfiles = append(consumedProfiles, namedProfile)
+
+	// delete account from remainingAccounts
+	remainingAccounts[accountIdx] = remainingAccounts[len(remainingAccounts)-1]
+	remainingAccounts = remainingAccounts[:len(remainingAccounts)-1]
+
+	return remainingAccounts, nil
+}
+
+func (c *completer) consumeProfiles(targetRoleName string, accounts []server.AWSAccount) ([]*AWSNamedProfile, []server.AWSAccount) {
+	consumedProfiles := []*AWSNamedProfile{}
+	remainingAccounts := []server.AWSAccount{}
+
+	for _, account := range accounts {
+		profileName := c.calculateDefaultProfileName(account)
+
+		// get the roles associated with this account
+		selectedProfile := server.AWSProfile{}
+		profiles := c.awsConfig.GetProfilesForAccount(account)
+		for _, profile := range profiles {
+			if strings.Contains(profile.RoleARN, targetRoleName) {
+				selectedProfile = profile
+			}
+		}
+
+		// if roleARN is populated, then we add it to the consumedProfiles list
+		if selectedProfile.RoleARN != "" {
+			currentProfile := &AWSNamedProfile{
+				Name:       profileName,
+				AWSProfile: selectedProfile,
+			}
+			consumedProfiles = append(consumedProfiles, currentProfile)
+			continue
+		}
+		// Else, then populate the remainingAccounts list
+		remainingAccounts = append(remainingAccounts, account)
+	}
 
-	return namedProfile, nil
+	return consumedProfiles, remainingAccounts
+}
+
+// SurveyRole will ask a user to configure a default role
+func (c *completer) SurveyRoles() ([]*AWSNamedProfile, error) {
+	// first prompt for roles
```

Comment:
I wonder if we should split these two modes of operations instead of "chaining them". Asking for a default role and then asking account/account what to do felt a bit confusing since we're kind of inverting the paradigm without warning.

Instead of trying to do both sequentially, what do you think about first asking the user to select between:
- Configuring the same role for every account that has it with our default generate profile name
- Configuring a particular role for a particular account with their particular profile name

Once the user picks one or the other we can essentially drop them into [this](https://github.com/chanzuckerberg/aws-oidc/pull/17/files#diff-b380719654262eb0e612e26c8974c831R159-R170) mode or [that](https://github.com/chanzuckerberg/aws-oidc/pull/17/files#diff-b380719654262eb0e612e26c8974c831R175-R189) mode depending on their answer?